src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs

```diff
@@ -363,12 +393,174 @@ public async Task produces_json_html_and_plain_text_formatted_values()
 
         var result = await kernel.SendAsync(new SubmitCode($"GET http://testuri.ninja"));
 
+        using var _ = new AssertionScope();
+
+        result.Events.Should().NotContainErrors();
+        result.Events.OfType<DisplayEvent>().Should().HaveCount(2);
+
+        var displayedValueProduced = result.Events.Should().ContainSingle<DisplayedValueProduced>().Which;
+        displayedValueProduced.FormattedValues.Should().ContainSingle().Which
+            .MimeType.Should().Be(HtmlFormatter.MimeType);
+
+        var returnValueProduced = result.Events.Should().ContainSingle<ReturnValueProduced>().Which;
+        var formattedValue = returnValueProduced.FormattedValues.Should().ContainSingle().Which;
+        formattedValue.MimeType.Should().Be(JsonFormatter.MimeType);
+        formattedValue.SuppressDisplay.Should().BeTrue();
+
+        displayedValueProduced.Value.Should().BeSameAs(returnValueProduced.Value);
+    }
+
+    [Fact]
+    public async Task produces_empty_response_for_slow_responses()
+    {
+        HttpRequestMessage request = null;
+        var slowHandler = new InterceptingHttpMessageHandler(async (message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            response.RequestMessage = message;
+            await Task.Delay(1500);
```

Comment:
Is this delay a magic number of some kind that indicates a "slow" response? Putting a hard-coded delay like this into a test slows tests down while also likely being nondeterministic. Let's work on improving this test design.

Reply:
The spinner is displayed on responses that take longer than a second. Yes glad to improve this if you have suggestions.